File: scripts/bootstrap/install-dependencies.pl
#!/usr/bin/env perl
use strict; use warnings; use File::Basename qw(dirname); use Cwd qw(abs_path); use File::Spec;
my $root=abs_path(File::Spec->catdir(dirname(__FILE__),'..','..')); chdir $root or die "Cannot enter $root: $!";
sub run { system(@_)==0 or die "Command failed: @_\n"; }
my $npm=$^O eq 'MSWin32' ? 'npm.cmd' : 'npm';
run('git','--version'); run($^X,'--version'); run($npm,'--version');
if(-f 'web/package-lock.json'){ run($npm,'ci'); } elsif(-f 'web/package.json'){ run($npm,'install'); }
if(-f 'python/requirements.txt'){
  my $py=$^X; run($py,'-m','venv','.venv') unless -d '.venv';
  my $venv=$^O eq 'MSWin32' ? '.venv/Scripts/python.exe' : '.venv/bin/python';
  run($venv,'-m','pip','install','-r','python/requirements.txt');
}
print "Bootstrap completed.\n";

File: scripts/bootstrap/install-dependencies.py
#!/usr/bin/env python3
"""Idempotent dependency bootstrap helper for CI and local development."""
import argparse, os, shutil, subprocess, sys
from pathlib import Path
ROOT=Path(__file__).resolve().parents[2]
def need(name):
    if not shutil.which(name): raise SystemExit(f"Missing required command: {name}")
def run(cmd, cwd=ROOT): print('+',' '.join(map(str,cmd))); subprocess.run(cmd,cwd=cwd,check=True)
def main():
    ap=argparse.ArgumentParser(); ap.add_argument('profile',nargs='?',default='developer'); args=ap.parse_args(); os.chdir(ROOT)
    for tool in ('git','node','npm', 'python' if os.name=='nt' else 'python3'): need(tool)
    venv=ROOT/'.venv'; py=venv/('Scripts/python.exe' if os.name=='nt' else 'bin/python')
    if not venv.exists(): run([sys.executable,'-m','venv',str(venv)])
    req=ROOT/'python/requirements.txt'
    if req.exists(): run([str(py),'-m','pip','install','--upgrade','pip']); run([str(py),'-m','pip','install','-r',str(req)])
    web=ROOT/'web';
    if (web/'package.json').exists(): run(['npm.cmd' if os.name=='nt' else 'npm','ci' if (web/'package-lock.json').exists() else 'install'],web)
    print('Bootstrap completed:',args.profile)
if __name__=='__main__': main()
